src/service/gif.js

```diff
@@ -0,0 +1,16 @@
+import api from './config';
+
+export const getGifFeatured = async () =>
+  new Promise((resolve, reject) => {
+    api
+      .get('/chat/gif/featured')
+      .then((res) => {
+        resolve(res.data);
+      })
+      .catch((err) => {
+        if (__DEV__) {
+          console.log('get gif featured: ', err);
+        }
+        reject(err);
+      });
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getGifFeatured` function is using promises in a way that could be simplified. Instead of creating a new Promise and resolving or rejecting it inside the `.then` and `.catch` blocks, you can return the promise returned by `api.get`. This will make your code cleaner and easier to understand.

```diff
- export const getGifFeatured = async () =>
-   new Promise((resolve, reject) => {
-     api
-       .get('/chat/gif/featured')
-       .then((res) => {
-         resolve(res.data);
-       })
-       .catch((err) => {
-         if (__DEV__) {
-           console.log('get gif featured: ', err);
-         }
-         reject(err);
-       });
-   });

+ export const getGifFeatured = async () => {
+   try {
+     const res = await api.get('/chat/gif/featured');
+     return res.data;
+   } catch (err) {
+     if (__DEV__) {
+       console.log('get gif featured: ', err);
+     }
+     throw err;
+   }
+ };
```
Also, consider adding error handling at a higher level where this function is called, rather than logging the error here and rethrowing it. This would follow the principle of separation of concerns, as this function's responsibility should be fetching data, not handling errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->